### PR TITLE
add act_cfg

### PR DESCRIPTION
**Pull request by xvjiarui**
This added config for activation so that user could use activation like  `relu` and `leaky_relu`.

### CODE REVIEW

File: mmdet/models/utils/activation.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import torch.nn as nn
+
+activation_cfg = {
+    # format: layer_type: (abbreviation, module)
+    'relu': nn.ReLU,
+    'leaky_relu': nn.LeakyReLU,
```
Comment (hellock) [inline mmdet/models/utils/activation.py]: May also add PReLU, RReLU, ReLU6, SELU, CELU, GELU.

File: mmdet/models/utils/activation.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+
+def build_activation_layer(cfg):
+    """ Build activation layer
+
+    Args:
+        cfg (dict): cfg should contain:
+            type (str): Identify activation layer type.
```
Comment (hellock) [inline mmdet/models/utils/activation.py]: Additional indents.

File: mmdet/models/utils/conv_module.py [hunk 3]
```diff
@@ -151,7 +150,7 @@ def norm(self):
         return getattr(self, self.norm_name)
 
     def init_weights(self):
-        nonlinearity = 'relu' if self.activation is None else self.activation
+        nonlinearity = self.act_cfg['type'] if self.with_activation else 'relu'
```
Comment (hellock) [inline mmdet/models/utils/conv_module.py]: nonlinearity can only be `relu` or `leaky_relu`.

File: mmdet/models/utils/activation.py (hunk 1 above)
Comment (xvjiarui) [reply mmdet/models/utils/activation.py]: `kaiming_init` and `inplace` option only support `relu` and `leaky_relu`.

Comment (hellock) [reply mmdet/models/utils/activation.py]: Yes, so we need to handle such cases in `init_weights`. Here we list all possibilities for potential usage (not necessarily in mmdet).

File: mmdet/models/utils/activation.py (hunk 2 above)
Comment (xvjiarui) [reply mmdet/models/utils/activation.py]: The indents are for the key-value in `cfg`.

File: mmdet/models/utils/activation.py (hunk 1 above)
Comment (xvjiarui) [reply mmdet/models/utils/activation.py]: All added except for GELU, which is not supported in pytorch 1.1

File: mmdet/models/bbox_heads/double_bbox_head.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def __init__(self,
             out_channels,
             kernel_size=1,
             bias=False,
-            activation=None,
+            act_cfg=None,
             conv_cfg=conv_cfg,
             norm_cfg=norm_cfg)
```
Comment (hellock) [inline mmdet/models/bbox_heads/double_bbox_head.py]: We may follow the argument order: conv_cfg, norm_cfg, act_cfg.

File: mmdet/models/necks/fpn_carafe.py [hunk 5]
```diff
@@ -52,7 +52,6 @@ def __init__(self,
         self.out_channels = out_channels
         self.num_ins = len(in_channels)
         self.num_outs = num_outs
-        self.activation = activation
```
Comment (hellock) [inline mmdet/models/necks/fpn_carafe.py]: We may also save act_cfg as a class attribute.

File: mmdet/models/utils/conv_module.py [hunk 6]
```diff
@@ -60,7 +61,7 @@ class ConvModule(nn.Module):
             False.
         conv_cfg (dict): Config dict for convolution layer.
         norm_cfg (dict): Config dict for normalization layer.
-        activation (str or None): Activation type, "ReLU" by default.
+        act_cfg (str or None): Activation type, "ReLU" by default.
```
Comment (hellock) [inline mmdet/models/utils/conv_module.py]: It is no longer a str.

File: mmdet/models/utils/activation.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import torch.nn as nn
+
+activation_cfg = {
+    # format: layer_type: (abbreviation, module)
```
Comment (hellock) [inline mmdet/models/utils/activation.py]: This comment is incorrect.

**Pull request merged**